sdk/python/feast/infra/online_stores/contrib/mysql_online_store/util.py

```diff
@@ -0,0 +1,325 @@
+    cur.execute(write_query, (table, feature_view_name, version_id, lock_row,))
+    if commit:
+        conn.commit()
+
+
+def _get_and_lock_feature_view_version(
+    cur: Cursor,
+    conn: Connection,
+    config: MySQLOnlineStoreConfig,
+    feature_view_name: str,
+) -> Optional[int]:
+    table, version_id = config.feature_version_table, None
+
+    read_query = "SELECT version_id, alter_in_progress FROM %s WHERE feature_view_name = %s FOR UPDATE;"
+    _execute_query_with_retry(
+        cur=cur,
+        conn=conn,
+        query=read_query,
+        values=(table, feature_view_name,),
+        commit=False,
+        propagate_exceptions=True
+    )
+    record = cur.fetchone()
+    if record is None:
+        logging.info(f'FeatureView ({feature_view_name} not found in {table}).')
+        conn.commit()
+        return record
+    version_id, alter_in_progress = record
+
+    # RB: obey the logical row lock in FeatureViewVersion table. this code path should be unlikely.
+    if alter_in_progress:
+        # unlock!
+        conn.commit()
+        raise ValueError(f'Unable to obtain row lock for table ({table}) and row ({feature_view_name}). ALTER '
+                         f'operation is in progress.')
+
+    lock_query = "UPDATE %s SET alter_in_progress = True FROM %s WHERE feature_view_name = %s;"
+    _execute_query_with_retry(
+        cur=cur,
+        conn=conn,
+        query=lock_query,
+        values=(table, feature_view_name,),
+        retries=20,
+        commit=False,
+        propagate_exceptions=True
+    )
+    conn.commit()
+    return version_id
+
+
+def _update_feature_view_version_id(
+    cur: Cursor,
+    conn: Connection,
+    config: MySQLOnlineStoreConfig,
+    feature_view_name: str,
+    new_version_id: int,
+) -> None:
+    table, version_id = config.feature_version_table, None
+
+    query = "SELECT alter_in_progress FROM %s WHERE feature_view_name = %s FOR UPDATE;"
+    _execute_query_with_retry(
+        cur=cur,
+        conn=conn,
+        query=query,
+        values=(table, feature_view_name,),
+        retries=5,
+        commit=False,
+        propagate_exceptions=True
+    )
+    record = cur.fetchone()
+    if record is None:
+        logging.info(f'FeatureView ({feature_view_name} not found in {table}).')
+        conn.commit()
+        return record
+
+    alter_in_progress = record
+
+    if not alter_in_progress:
+        conn.commit()
+        raise ValueError(f'Invalid operation: table ({table}) cannot be updated for FeatureView ({feature_view_name})'
+                         f'without alter_in_progress being set to True.')
+
+    lock_query = "UPDATE %s SET version_id = %s, alter_in_progress = False FROM %s WHERE feature_view_name = %s;"
+    _execute_query_with_retry(
+        cur=cur,
+        conn=conn,
+        query=lock_query,
+        values=(table, new_version_id, feature_view_name,),
+        retries=20,
```

Comment:
nit, this should probably be configured more explicitly somewhere but not a huge deal.